src/Apps/Artwork/Components/ArtworkSidebar/index.tsx

```diff
@@ -80,3 +85,48 @@ export const ArtworkSidebarFragmentContainer = createFragmentContainer(
     }
   `
 )
+
+export const ArtworkSidebarQueryRenderer = ({
+  artworkID,
+}: {
+  artworkID: string
+}) => {
+  return (
+    <ContextConsumer>
+      {({ user, mediator, relayEnvironment }) => {
+        return (
+          <QueryRenderer<ArtworkSidebarQuery>
+            environment={relayEnvironment}
+            variables={{ artworkID }}
+            query={graphql`
+              query ArtworkSidebarQuery($artworkID: String!) {
+                artwork(id: $artworkID) {
+                  ...ArtworkSidebar_artwork
+                }
+              }
+            `}
+            render={({ props }) => {
+              if (props) {
+                return (
+                  <ArtworkSidebarFragmentContainer
+                    artwork={props.artwork as any}
+                  />
+                )
+              } else {
+                return (
+                  <SpinnerContainer
+                    width="100%"
+                    height="100px"
+                    position="relative"
+                  >
```

Comment:
Can these props be passed directly to `<Spinner>`? And if not, it might be worthwhile to update spinner in pallete to have a container.

Reply:
Please use this function instead, as it will make sure tooling like renderRelayTree works, we can add error/retry handling in one place: https://github.com/artsy/reaction/blob/master/src/Artsy/Relay/renderWithLoadProgress.tsx